### PR TITLE
SIMD implementations for min/max and rounding blocks

### PR DESCRIPTION
**Pull request by ncorgan**

### CODE REVIEW

**Comment by ncorgan** [comment]
Hold off for now, cleanup of platform support

**Comment by ncorgan** [comment]
Fixed

File: common/XSIMDTypes.hpp [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+// This header is to be included by XSIMD implementation files. The
+// compile flags used for the file will determine which of these
+// macros will be defined. This allows us to use the SFINAE structs
+// below to only build SIMD implementations for types supported on the
+// given platform. If the type isn't supported, a fallback implementation
+// will be used.
+//
+// See: https://xsimd.readthedocs.io/en/latest/api/available_wrappers.html
+//
+
```
Comment (guruofquality) [inline common/XSIMDTypes.hpp:22]: Should PothosCore provided this? Looks like type detection that would be generally useful?

Comment (ncorgan) [reply common/XSIMDTypes.hpp:22]: I figured this was a detail of using XSIMD for the SIMD stuff, but with this being common (the same file is probably in PothosComms), I can see the argument for it. I'll move it over with some tweaks and add a "can CMake find this header" condition for building the relevant modules.

**Review by guruofquality — APPROVED** [review]

**Pull request merged**